src/ra_log_reader.erl

```diff
@@ -95,18 +101,30 @@ close(#?STATE{open_segments = Open}) ->
 -spec update_segments([segment_ref()], state()) -> state().
 update_segments(NewSegmentRefs,
                 #?STATE{open_segments = Open0,
-                        segment_refs = SegmentRefs0} = State) ->
-    SegmentRefs = compact_seg_refs(NewSegmentRefs, SegmentRefs0),
+                        segment_refs = SegRefs0} = State) ->
+
+    SegmentRefs0 = ra_lol:to_list(SegRefs0),
+    SegmentRefsComp = compact_segrefs(NewSegmentRefs, SegmentRefs0),
+    SegmentRefsCompRev = lists:reverse(SegmentRefsComp ),
+    SegRefs = ra_lol:from_list(fun seg_ref_gt/2, SegmentRefsCompRev),
+    Range = case SegmentRefsComp of
+                  [{{_, L}, _} | _] ->
+                      [{{F, _}, _} | _] = SegmentRefsCompRev,
+                      ra_range:new(F, L);
+                  _ ->
+                      undefined
+              end,
```

Comment:
minor formatting thing: I think this block needs one or two columns of indent removed